README.md

```diff
@@ -7388,105 +7387,128 @@ See [SQLCipher 4.0.0 Release](https://www.zetetic.net/blog/2018/11/30/sqlcipher-
+import GRDB
+import SQLCipher
 
-try clearDBQueue.inDatabase { db in
-    try db.execute(sql: "ATTACH DATABASE ? AS encrypted KEY ?", arguments: [encryptedDBQueue.path, "secret"])
-    try db.execute(sql: "SELECT sqlcipher_export('encrypted')")
-    try db.execute(sql: "DETACH DATABASE encrypted")
+var config = Configuration()
+config.prepareDatabase = { db in
+    ... // Carefully load passphrase bytes
+    let code = sqlite3_key(db.sqliteConnection, /* passphrase bytes */)
+    ... // Carefully dispose passphrase bytes
+    guard code == SQLITE_OK else {
+        throw DatabaseError(resultCode: code, message: db.lastErrorMessage)
+    }
 }
-
-// Now the copy is done, and the clear-text database can be deleted.
+let dbQueue = try DatabaseQueue(path: dbPath, configuration: config)
 ```
 
-### Attempting Several Passphrases
+#### Passphrase availability vs. Database availability
 
-TODO. Context: synchronizing the passphrase in the keychain, and the passphrase used by the database is tricky: one saving operation may succeed, and the other fail. It is possible that one has to try *two* passphrases:
+When the passphrase is securely stored in the system keychain, your application can protect is using the [`kSecAttrAccessible`](https://developer.apple.com/documentation/security/ksecattraccessible) attribute.
 
 ```swift
 var config = Configuration()
 config.prepareDatabase = { db in
-    do {
-        try db.usePassphrase(passphrase1)
-    } catch let error as DatabaseError where error.resultCode == .SQLITE_NOTADB {
-        try db.usePassphrase(passphrase2)
-    }
+    let passphrase = try loadPassphraseFromSystemKeychain()
+    try db.usePassphrase(passphrase)
 }
 ```
 
+Such protection prevents GRDB from creating SQLite connections when the passphrase is not available:
 
-### Security Considerations
+```swift
+// Success if and only if the passphrase is available
+let dbQueue = try DatabaseQueue(path: dbPath, configuration: config)
+```
+
+For the same reason, [database pools](#database-pools), which open SQLite connections on demand, may fail at any time as soon as the passphrase becomes unavailable:
+
+```swift
+// Success if and only if the passphrase is available
+let dbPool = try DatabasePool(path: dbPath, configuration: config)
+
+// May fail if passphrase has turned unavailable
+try dbPool.read { ... }
 
-- TODO: Connections can be used after passphrase has turned unavailable
-- TODO: Advanced passphrase protection (use cases for the raw C SQLCioher functions)
+// May trigger value observation failure if passphrase has turned unavailable
+try dbPool.write { ... }
+```
+
+Those eventual database pool failures are not predictable, because pools maintain a pool of long-lived SQLite connection(s). Some database accesses will use those existing connections, and succeed. Some other database accesses will fail, as soon as the pool wants to open a new connection.
```

Comment:
> pools maintain a pool of long-lived

maybe less ambiguous:

> DatabasePool maintains a pool of long-lived SQLite connection(s)

Reply:
You're right, this is very bad English. I did attempt a full rewrite of the paragraph: cc7d19d92567c6c71ba80ba474db7df5fc72afe6